### ISSUE TITLE
Do you think the web development industry is hostile to newcomers?

### ISSUE THREAD

**Issue by btpschroeder** [taniarascia/ama#5]
And if so, how would you propose making our industry more welcoming and inclusive  of those who are potentially interested in the field? (This can involve politics e.g. gender/race/sexuality, or more simple stuff like "amount of tutorials available for newcomers" -- just interested to hear your thoughts in general since I know this is something you're passionate about.)

**Comment by taniarascia** [comment]
Great question, Ben! I definitely do think the web development industry is hostile to newcomers. Once you've been around for a while, you get used to it and learn "the right way" to ask questions and navigate through communities, and it can be difficult to remember what it was like to be a newcomer again.

Regarding politics, I can't speak too much on that, since anonymity is still easily possible to achieve. Whether or not it's harder to get a job based on your race/gender/sexuality, the difficulties that newcomers face in getting help, finding resources, and joining communities is across the board.

The difficult thing about web development is knowing where to start. There are endless resources, languages, frameworks, buzzwords, tutorials, and opinions floating around the web, all from different people with different teaching styles and different assumptions over the past 20 years. It's only logical that a newcomer would want to go on a forum and ask a question about where to go or what to do, but such broad questions are usually ignored or answered with useless suggestions.

The new developer will be told to "start small, and make bigger projects as you go". I believe this is pretty much useless advice, as a new developer might not even have a clue yet about syntax and terminology and setting up an environment, and can not reasonably be expected to come up with their own solutions to solve. How can you know, when you hardly know anything at all, whether or not your problem is simple to solve or exceedingly complex?

I don't think very many people make much money from teaching development, and they certainly can't make as much as an educator than as an actual developer or programmer. This leads to most resources being put out there by the goodness of someone's heart if they have time, but even if you find an amazing tutorial that matches your learning style by one of these people, you can't be guaranteed that they've written anything else. 

Asking questions for a newcomer is exceedingly difficult. Any developer who has spend any amount of time coding knows the intricate dance they must do to get a question even accepted by Stack Overflow. I know the first few I tried to post immediately got removed for being a duplicate, or got an elitist, angry response ("Have you ever heard of functions!?") You can yell at a newbie for not reading 10 pages of rules before they post a question, but honestly - who ever has?

With the [curse of knowlegde](https://en.wikipedia.org/wiki/Curse_of_knowledge), it's hard for us to understand where newcomers are coming from. Sometimes, it may seem as if they're asking questions about things as obvious as breathing. It's important to maintain compassion and understanding and respect for other people. If you don't want to help, simply ignore them. If you do want to help, genuinely help and refrain from being elitist or snarky. 

For those (such as myself) who create resources that are intended to be useful to new developers, providing concrete resources with a beginning and an end, and clear prerequisites and goals is essential. Nothing should be assumed. 

So, to summarize - be compassionate, be helpful, and always keep in mind the frustrations you experienced when learning, remembering that there is more to learn now than ever.
